src/main/java/com/synopsys/integration/blackduck/installer/configure/BlackDuckWaitJobTask.java

```diff
@@ -0,0 +1,90 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.synopsys.integration.blackduck.installer.configure;
+
+import com.synopsys.integration.blackduck.api.generated.discovery.ApiDiscovery;
+import com.synopsys.integration.blackduck.api.generated.response.CurrentVersionView;
+import com.synopsys.integration.blackduck.configuration.BlackDuckServerConfig;
+import com.synopsys.integration.blackduck.installer.exception.BlackDuckInstallerException;
+import com.synopsys.integration.blackduck.installer.exception.IntegrationKeyStoreException;
+import com.synopsys.integration.blackduck.service.BlackDuckService;
+import com.synopsys.integration.blackduck.service.BlackDuckServicesFactory;
+import com.synopsys.integration.exception.IntegrationException;
+import com.synopsys.integration.log.IntLogger;
+import com.synopsys.integration.wait.WaitJobTask;
+import org.apache.commons.lang3.StringUtils;
+
+import javax.net.ssl.SSLHandshakeException;
+import java.io.File;
+
+public class BlackDuckWaitJobTask implements WaitJobTask {
+    private final IntLogger intLogger;
+    private final BlackDuckServerConfig blackDuckServerConfig;
+    private final UpdateKeyStoreService updateKeyStoreService;
+    private final File installDirectory;
+
+    public BlackDuckWaitJobTask(IntLogger intLogger, BlackDuckServerConfig blackDuckServerConfig, UpdateKeyStoreService updateKeyStoreService, File installDirectory) {
+        this.intLogger = intLogger;
+        this.blackDuckServerConfig = blackDuckServerConfig;
+        this.updateKeyStoreService = updateKeyStoreService;
+        this.installDirectory = installDirectory;
+    }
+
+    @Override
+    public boolean isComplete() throws BlackDuckInstallerException {
+        BlackDuckServicesFactory blackDuckServicesFactory = blackDuckServerConfig.createBlackDuckServicesFactory(intLogger);
+        BlackDuckService blackDuckService = blackDuckServicesFactory.createBlackDuckService();
+        try {
+            CurrentVersionView currentVersionView = blackDuckService.getResponse(ApiDiscovery.CURRENT_VERSION_LINK_RESPONSE);
+            if (null != currentVersionView) {
+                String currentVersion = currentVersionView.getVersion();
+                if (StringUtils.isNotBlank(currentVersion)) {
+                    intLogger.info(String.format("Black Duck server found running version %s.", currentVersion));
+                    return true;
+                }
+            }
+        } catch (IntegrationException e) {
+            if (e.getCause() instanceof SSLHandshakeException) {
+                intLogger.info("The Black Duck server is responding, but its certificate is not in the java keystore.");
```

Comment:
Should this log message be changed to warn or error? And should we throw an exception here so it wont continue trying. If the certificate is missing, Im not sure why we would continue trying to connect.

Reply:
I'll change to warn